e3sm_to_cmip/lib.py

```diff
@@ -735,8 +735,14 @@ def load_axis(data, levels=None, has_time=True):
 
         axis_bnds = levels.get("e3sm_axis_bnds")
         if axis_bnds:
+            cell_bounds = data[axis_bnds]
+            # i.g. handler cl, cli, clw returns xarray dataarray
+            if not isinstance(data[axis_bnds], np.ndarray):
+                cell_bounds = data[axis_bnds].values
```

Comment:
@tomvothecoder I saw error for `tsl` handler as follows
`  File "/home/zhang40/miniconda3/envs/e2c_nco511_13/lib/python3.9/site-packages/e3sm_to_cmip/lib.py", line 741, in load_axis
    cell_bounds = data[axis_bnds].values
'list' object has no attribute 'values'`
It looks like the type checker needs to be extended to `list` as well, or it is better to fix the handler `tsl`?

Reply:
The logic in `load_axis()`:
https://github.com/E3SM-Project/e3sm_to_cmip/blob/4ab88cd6fe3d7ac58b7e70dac15dde9f1217eb5e/e3sm_to_cmip/lib.py#L736-L742

Background Info on logic:

* `data` is a dictionary that stores components of the `tsl` variable such as coordinates, bounds, etc.
* `axis_bnds` simply refers to `"e3sm_axis_bnds"`
* `tsl` handler defines `"e3sm_axis_bnds"` below as a single string (`"levgrnd_bnds"`)
https://github.com/E3SM-Project/e3sm_to_cmip/blob/4dcf50676733d7ed3123a6efca7d9f9e1b3170e3/e3sm_to_cmip/cmor_handlers/tsl.py#L21

What we expect:

* `axis_bnds` = `"e3sm_axis_bnds"` =  `"levgrnd_bnds"` 
* `data[axis_bnds]` should return an `xarray.DataArray` representing `"levgrnd_bnds"`

What is actually happening:

`data["axis_bnds"]` is returning a `list` instead of an `xarray.DataArray`.

**The question here is why is `data["axis_bnds"]` returning a `list` if there is only a single string `"e3sm_axis_bnds"` defined?** 

Can you provide the command/script you ran for `tsl` specifically?

Reply:
This is the section of code that sets `data[axis_bnds]`:

https://github.com/E3SM-Project/e3sm_to_cmip/blob/4dcf50676733d7ed3123a6efca7d9f9e1b3170e3/e3sm_to_cmip/lib.py#L707-L713

Reply:
I think I figured it out.

Using this command:
```
e3sm_to_cmip --realm lnd -i /p/user_pub/e3sm/zhang40/cmorized_for_testing/rgr_lnd/historical_r1i1p1f1 -o /p/user_pub/e3sm/zhang40/cmorized_for_testing  -v tsl -t /home/zhang40/cmip6-cmor-tables/Tables/ -u /home/zhang40/CMIP6-Metadata/E3SM-2-0/historical_r1i1p1f1.json
```

I added some print statements and noticed `data[axis_bnds]` was a list of floats. This means it is already in the correct format needed for `cmor.write()` (unless it needs to by a `numpy` array).

```
[*] Loaded `tsl` handler for variable `tsl`.
[*] Writing log output to: /p/user_pub/e3sm/zhang40/cmorized_for_testing/converter.log
[+] Running CMOR handlers in parallel
  0%|                                           | 0/1 [00:00<?, ?it/s]

CELL_BOUNDS:
[0, 0.01751106046140194, 0.045087261125445366, 0.09055273048579693, 0.16551261954009533, 0.28910057805478573, 0.4928626772016287, 0.8288095649331808, 1.3826923426240683, 2.2958906944841146, 3.801500206813216, 6.28383076749742, 10.376501685008407, 17.124175196513534, 28.249208575114608, 42.098968505859375]

Problem with 'cmor.write'. Please check the logfile (if defined).
100%|███████████████████████████████████| 1/1 [00:01<00:00,  1.12s/it]
[+] 1 of 1 handlers complete
(e3sm_to_cmip_dev) 
```

I just pushed a commit to fix the conditionals. It looks like there is a new, ambiguous error message being thrown: `Problem with 'cmor.write'. Please check the logfile (if defined).`.

This has to do with the `write_data()` function in `tsl` here:
https://github.com/E3SM-Project/e3sm_to_cmip/blob/4dcf50676733d7ed3123a6efca7d9f9e1b3170e3/e3sm_to_cmip/cmor_handlers/tsl.py#L25-L35

Reply:
It looks like there is a new, ambiguous error message being thrown: `Problem with 'cmor.write'. Please check the logfile (if defined).`.

I think this has to do with the `write_data()` function in `tsl` here:
https://github.com/E3SM-Project/e3sm_to_cmip/blob/4dcf50676733d7ed3123a6efca7d9f9e1b3170e3/e3sm_to_cmip/cmor_handlers/tsl.py#L25-L35

Reply:
> It looks like there is a new, ambiguous error message being thrown: `Problem with 'cmor.write'. Please check the logfile (if defined).`.

For future reference, I checked the `cmor` log for the specific error.
```
Error: The source_id, "E3SM-2-0", found in your 
! input file (/p/user_pub/e3sm/zhang40/cmorized_for_testing/user_metadata.json) could not be found in 
! your Controlled Vocabulary file. (/home/vo13/PCMDI/cmip6-cmor-tables/Tables//CMIP6_CV.json) 
```

The fix was to run `git pull` on `cmip6-cmor-tables` repo. This is another example of #124 being useful because it can be easy to forget to pull the latest changes from an external repo.